### PR TITLE
feat: support `@mel.new` in `@mel.send` and `@mel.send.pipe`

### PR DESCRIPTION
**Pull request by anmonteiro**
fixes https://github.com/melange-re/melange/issues/757

### CODE REVIEW

File: jscomp/core/lam_compile_external_call.ml [hunk 1]
```diff
@@ -312,18 +312,24 @@ let translate_ffi (cxt : Lam_compile_context.t) arg_types
...
             (let self = translate_scoped_access js_send_scopes self in
-             E.call
-               ~info:{ arity = Full; call_info = Call_na }
-               (E.dot self name) args)
+             match new_ with
+             | true -> E.new_ (E.dot self name) args
+             | false ->
+                 E.call
+                   ~info:{ arity = Full; call_info = Call_na }
+                   (E.dot self name) args)
```
Comment (jchavarri) [inline jscomp/core/lam_compile_external_call.ml:332]: Do you think it's worth to abstract this `match` into a function? The same code appears 4 times.

File: ppx/ast_external_process.ml [hunk 2]
```diff
@@ -826,16 +826,31 @@ let external_desc_of_non_obj (loc : Location.t) (st : external_desc)
...
+      | _ :: _, `Nm_na ->
+          Js_send
+            {
+              splice;
+              name;
+              js_send_scopes = scopes;
+              pipe = false;
+              new_ = false;
+            }
+      | _ :: _, (`Nm_val _ | `Nm_external _) ->
```
Comment (jchavarri) [inline ppx/ast_external_process.ml:850]: `Nm_external` can't be produced with "new", right?

**Review by jchavarri — COMMENTED** [review]
This is great, thanks!

File: jscomp/core/lam_compile_external_call.ml (hunk 1 above)
Comment (anmonteiro) [reply jscomp/core/lam_compile_external_call.ml:332]: absolutely

File: ppx/ast_external_process.ml (hunk 2 above)
Comment (anmonteiro) [reply ppx/ast_external_process.ml:850]: `Nm_payload` can't, but `Nm_external` can.

**Review by jchavarri — APPROVED** [review]

**Comment by anmonteiro** [comment]
forgot a changelog entry for this one

**Pull request merged**